File: lib/features/files/file_service.dart
import '../../core/models/models.dart';
import '../../core/network/api_client.dart';

/// Application service for file-manager operations.
/// It contains no widget or filesystem code; all remote access stays in ApiClient.
class FileService {
  const FileService(this._api);

  final ApiClient _api;

  Future<List<ManagedFile>> list({
    String? search,
    String sort = 'date',
    bool descending = true,
  }) => _api.files(search: search, sort: sort, descending: descending);

  Future<ManagedFile> rename(String id, String filename) =>
      _api.renameFile(id, filename);

  Future<ManagedFile> move(String id, String folder) =>
      _api.moveFile(id, folder);

  Future<ManagedFile> open(String id) => _api.openFile(id);

  Future<ManagedFile> share(String id) => _api.shareFile(id);

  Future<void> delete(String id) => _api.deleteFile(id);

  Future<void> favorite(String id, bool value) => _api.setFavorite(id, value);
}

File: backend/app/services/file_service.py
from uuid import UUID

from app.repositories.library import LibraryRepository
from app.services.files import FileManagerService
from app.services.storage import StorageService
from app.schemas.files import FileActionResponse, FileInfoResponse


class FileService:
    def __init__(self, repository: LibraryRepository, storage: StorageService | None = None) -> None:
        self.manager = FileManagerService(repository, storage=storage)

    def list(self, user_id: UUID, *, search: str | None, sort: str, descending: bool):
        return self.manager.list(user_id, search=search, sort=sort, descending=descending)

    def info(self, user_id: UUID, item_id: UUID) -> FileInfoResponse | None:
        return self.manager.info(user_id, item_id)

    def rename(self, user_id: UUID, item_id: UUID, filename: str) -> FileActionResponse | None:
        return self.manager.rename(user_id, item_id, filename)

    def move(self, user_id: UUID, item_id: UUID, folder: str) -> FileActionResponse | None:
        return self.manager.move(user_id, item_id, folder)

    def delete(self, user_id: UUID, item_id: UUID) -> FileActionResponse | None:
        return self.manager.delete(user_id, item_id)

    def open(self, user_id: UUID, item_id: UUID) -> FileInfoResponse | None:
        return self.manager.open(user_id, item_id)

    def share(self, user_id: UUID, item_id: UUID) -> FileInfoResponse | None:
        return self.manager.share(user_id, item_id)
